packages/clerk-js/src/core/resources/SignUp.ts

```diff
@@ -167,7 +167,8 @@ export class SignUp extends BaseResource implements SignUpResource {
 
   public authenticateWithRedirect = async (params: AuthenticateWithRedirectParams): Promise<void> => {
     const { redirectUrl, redirectUrlComplete, strategy } = params || {};
-    const { verifications } = await this.create({
+    const authenticateFn = this.id ? this.update : this.create;
```

Comment:
🙃 
I'd write this as:
```suggestion
    const authenticateFn = (...args) => this.id ? this.update(...args) : this.create(...args);
```
`authenticateFn` does not run with the same context (`this`) as `this.update` and `this.create` do. 

Note that in this particular case, it's totally fine because `create` and `update` are declared as arrow functions so they do retain their context even if called without it (eg: `this.create` vs `create()`) but if the declaration of these function changes in a future refactor, this could break